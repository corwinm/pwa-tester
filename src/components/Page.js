import React from "react";
import ErrorBoundary from "components/ErrorBoundary";

export default function Page({ title, children }) {
  return (
    <ErrorBoundary>
      <section className="section">
        <div className="container is-fluid">
          <h1 className="title has-text-white">{title}</h1>
          <div>{children}</div>
        </div>
      </section>
    </ErrorBoundary>
  );
}
