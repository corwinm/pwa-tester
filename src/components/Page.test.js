import React from "react";
import { render, cleanup } from "react-testing-library";
import Page from "./Page";

afterEach(cleanup);

test("matches snapshot", () => {
  const { container } = render(<Page />);
  expect(container.firstChild).toMatchSnapshot();
});

test("renders title", () => {
  const expectedTitle = "Expected Title";
  const { container, getByText } = render(<Page title={expectedTitle} />);

  expect(container.firstChild).toMatchSnapshot();
  expect(getByText(expectedTitle)).toBeTruthy();
});

test("renders children", () => {
  const expectedChildren = "Expected Children";
  const { container, getByText } = render(<Page>{expectedChildren}</Page>);

  expect(container.firstChild).toMatchSnapshot();
  expect(getByText(expectedChildren)).toBeTruthy();
});

test("renders title and children", () => {
  const expectedTitle = "Expected Title";
  const expectedChildren = "Expected Children";
  const { container, getByText } = render(
    <Page title={expectedTitle}>{expectedChildren}</Page>
  );

  expect(container.firstChild).toMatchSnapshot();
  expect(getByText(expectedTitle)).toBeTruthy();
  expect(getByText(expectedChildren)).toBeTruthy();
});

test("renders fallback UI for errors", () => {
  const expectedErrorMessage = "Expected Error Message";
  const ErrorThrowingChild = () => {
    throw Error(expectedErrorMessage);
  };
  const { container, getByText } = render(
    <Page>
      <ErrorThrowingChild />
    </Page>
  );

  expect(container).toMatchSnapshot();
  expect(getByText(expectedErrorMessage)).toBeTruthy();
});
