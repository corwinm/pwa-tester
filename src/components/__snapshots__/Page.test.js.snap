// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="notch-inset-item"
>
  <section
    class="section"
  >
    <div
      class="container is-fluid"
    >
      <h1
        class="title has-text-white"
      />
      <div />
    </div>
  </section>
</div>
`;

exports[`renders children 1`] = `
<div
  class="notch-inset-item"
>
  <section
    class="section"
  >
    <div
      class="container is-fluid"
    >
      <h1
        class="title has-text-white"
      />
      <div>
        Expected Children
      </div>
    </div>
  </section>
</div>
`;

exports[`renders fallback UI for errors 1`] = `
<div>
  <h1>
    Something went wrong.
  </h1>
  <p>
    Error
  </p>
  <p>
    Expected Error Message
  </p>
  <p>
    <button
      class="button"
    >
      Reload Page
    </button>
  </p>
</div>
`;

exports[`renders title 1`] = `
<div
  class="notch-inset-item"
>
  <section
    class="section"
  >
    <div
      class="container is-fluid"
    >
      <h1
        class="title has-text-white"
      >
        Expected Title
      </h1>
      <div />
    </div>
  </section>
</div>
`;

exports[`renders title and children 1`] = `
<div
  class="notch-inset-item"
>
  <section
    class="section"
  >
    <div
      class="container is-fluid"
    >
      <h1
        class="title has-text-white"
      >
        Expected Title
      </h1>
      <div>
        Expected Children
      </div>
    </div>
  </section>
</div>
`;
